src/OrchardCore.Modules/OrchardCore.Localization/Drivers/LocalizationSettingsDisplayDriver.cs

```diff
@@ -0,0 +1,101 @@
+using System;
+using System.Globalization;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Mvc.Localization;
+using Microsoft.Extensions.Localization;
+using Newtonsoft.Json;
+using OrchardCore.DisplayManagement.Entities;
+using OrchardCore.DisplayManagement.Handlers;
+using OrchardCore.DisplayManagement.Notify;
+using OrchardCore.DisplayManagement.Views;
+using OrchardCore.Environment.Shell;
+using OrchardCore.Localization.ViewModels;
+using OrchardCore.Settings;
+
+namespace OrchardCore.Localization.Drivers
+{
+    public class LocalizationSettingsDisplayDriver : SectionDisplayDriver<ISite, LocalizationSettings>
+    {
+        public const string GroupId = "localization";
+
+        private readonly INotifier _notifier;
+        private readonly IShellHost _shellHost;
+        private readonly ShellSettings _shellSettings;
+
+        public LocalizationSettingsDisplayDriver(
+            INotifier notifier,
+            IShellHost shellHost,
+            ShellSettings shellSettings,
+            IHtmlLocalizer<LocalizationSettingsDisplayDriver> h,
+            IStringLocalizer<LocalizationSettingsDisplayDriver> s
+        )
+        {
+            _notifier = notifier;
+            _shellHost = shellHost;
+            _shellSettings = shellSettings;
+            H = h;
+            S = s;
+        }
+
+        IHtmlLocalizer H { get; set; }
+        IStringLocalizer S { get; set; }
+
+        public override IDisplayResult Edit(LocalizationSettings section, BuildEditorContext context)
+        {
+            return Initialize<LocalizationSettingsViewModel>("LocalizationSettings_Edit", model =>
+                {
+                    model.Cultures = CultureInfo.GetCultures(CultureTypes.AllCultures)
```

Comment:
Can we cache all cultures instead on calling each time

Reply:
Not necessary. It's called rarely (only when editing the cultures) and it won't be collectable then.